### PR TITLE
Fix/joining room using os.join room() fails if casual os does not have permission to camera or mic

### LINKED ISSUE #568: Joining a room using `os.joinRoom()` fails if CasualOS doesn't have permission to access camera or microphone
Instead, it just silently fails to establish a WebRTC connection. CasualOS should ask for camera and microphone permission first, before trying to join a room.

### PR DESCRIPTION
**Pull request by TroyceGowdy**
fixes #568

### CODE REVIEW

File: src/aux-vm-browser/managers/LivekitManager.ts [hunk 1]
```diff
@@ -98,9 +98,24 @@ export class LivekitManager implements SubscriptionLike {
...
+            await navigator.mediaDevices.getUserMedia({
+                video: true,
+                audio: true,
+            });
+            console.log('[LivekitManager] Media permissions granted.');
+        } catch (error) {
+            console.error('[LivekitManager] Media permissions denied:', error);
+            throw new Error('Media permissions are required to join the room.');
+        }
+    }
```
Comment (KallynGowdy) [inline src/aux-vm-browser/managers/LivekitManager.ts:113]: @TroyceGowdy Do you know if this error is able to be handled by the CasualOS script that called `os.joinRoom()`?

Comment (KallynGowdy) [reply src/aux-vm-browser/managers/LivekitManager.ts:113]: Ok, but would the script be able to get the error and automatically check that the user didn't grant permission?

Comment (TroyceGowdy) [reply src/aux-vm-browser/managers/LivekitManager.ts:113]: Yes it does. Just checked the toast message and it is coming through.

Comment (KallynGowdy) [reply src/aux-vm-browser/managers/LivekitManager.ts:113]: Ok, good. Also one more question: if the user has already granted permissions, does it prompt for permissions again or does it just join the room?

Comment (TroyceGowdy) [reply src/aux-vm-browser/managers/LivekitManager.ts:113]: It just joins the room

**Pull request merged**